src/miner.cpp

```diff
@@ -568,7 +569,12 @@ void BitcoinMiner(CWallet* pwallet, bool fProofOfStake)
     const int64_t nSpacingMillis = consensus.nTargetSpacing * 1000;
 
     // Each thread has its own key and counter
-    CReserveKey reservekey(pwallet);
+    Optional<CReserveKey> opReservekey{nullopt};
```

Comment:
Nit: should probably include our `optional.h` wrapper directly in the header includes rather than implicitly